src/components/radio-group/radio-group-component.jsx

```diff
@@ -1,113 +1,68 @@
-const RICHNESS = 'richness';
-
-const RadioGroup = ({ activeLayers, options, title, handleSimpleLayerToggle, handleExclusiveLayerToggle, handleInfoClick, isFullscreenActive }) => {
-
-  const optionsLayers = [];
-  options.forEach(option => {
-    Object.keys(option.layers).forEach(variant => {
-      optionsLayers.push({
-        layerId: option.layers[variant],
-        variant,
-        option
-      });
-    });
-  });
-  const selectedLayersIds = activeLayers.map(l => l.title);
-  const selected = optionsLayers.find(o => selectedLayersIds.includes(o.layerId));
-  const selectedLayer = selected && selected.layerId;
-  const variant = (selected && selected.variant) || RARITY;
-  const isRarityActive = variant === RARITY;
-
-  const isOnMobile = useMobile();
-
-  const isSelected = (option) => !!(activeLayers.find(l => l.title === option.layers[variant]));
-  const isLastSelected = (option) => {
-    const firstSelectedLayer = activeLayers.find(layer => layer.category === "Biodiversity");
-    return firstSelectedLayer && firstSelectedLayer.title === option.layers[variant];
-  }
+const getSelectedLayerId = (options) => {
+  const selectedLayer = options && options.find((o) => o.selected);
+  return selectedLayer && selectedLayer.layer;
+}
 
-  const renderRadioButton = option => {
-    const isOptionSelected = isSelected(option);
+const RadioGroup = ({
+  radioOptions,
+  title,
+  handleSimpleLayerToggle,
+  handleExclusiveLayerToggle,
+  handleInfoClick,
+  variant
+}) => {
+  const selectedLayerId = useMemo(() => getSelectedLayerId(radioOptions), [radioOptions]);
+  const renderRadioButton = (option) => {
+    const key = `radio-button-${title}-${option.value}-${variant}`;
     return (
-      <div key={option.value} className={cx(
-        styles.radioOption,
-        { [styles.radioOptionSelected]: isOptionSelected }
-      )}>
+      <div
+        key={key}
+        className={cx(styles.radioOption, {
+          [styles.radioOptionSelected]: option.selected
+        })}
+      >
         <RadioButton
-          key={option.value}
           name={title}
           value={option.value}
-          checked={isOptionSelected}
+          checked={option.selected}
+          id={key}
           onClick={() => {
-            if (isOptionSelected || !selectedLayer) {
-              handleSimpleLayerToggle(option.layers[variant], isOptionSelected);
+            if (option.selected || !selectedLayerId) {
+              handleSimpleLayerToggle(option.layer, option.selected);
             } else {
-              handleExclusiveLayerToggle(option.layers[variant], selectedLayer);
+              handleExclusiveLayerToggle(option.layer, selectedLayerId);
             }
           }}
           text={option.name}
         />
-        {isOptionSelected && (
+        {option.selected && (
           <div className={styles.toggle}>
             <InfoIcon
               className={styles.icon}
               onClick={() => handleInfoClick(option, variant)}
```

Comment:
It seems that the `variant` is not reaching the component. It states `undefined` on the metadata modal

![image](https://user-images.githubusercontent.com/6906348/109933487-c0d01600-7ccb-11eb-8e66-a66c9f06ff09.png)